core/services/ping/pingdriver.py

```diff
@@ -0,0 +1,58 @@
+import logging
+from typing import Optional
+
+from brping import PingDevice
+from brping.definitions import COMMON_DEVICE_INFORMATION
+
+from bridges import Bridges
+from pingutils import PingDeviceDescriptor
+from serialhelper import Baudrates, set_low_latency, set_sensor_baudrate
+
+
+class PingDriver:
+    def __init__(self, ping: PingDeviceDescriptor, port: int) -> None:
+        self.ping = ping
+        self.port = port
+        self.bridge: Optional[Bridges] = None
+
+    def detect_highest_baud(self) -> Baudrates:
+        """Tries to communicate in increasingly high baudrates up to 4M
+        returns the highest one with at least 90% success rate.
+        """
+        failure_threshold = 0.1  # allow up to 10% failure rate
+        attempts = 10  # try up to 10 times per baudrate
+        max_failures = attempts * failure_threshold
+
+        last_valid_baud = Baudrates.b115200
+        for baud in Baudrates:
+            # Open device with 2M baud to setup Ping360
+            logging.debug(f"Trying baud {baud}...")
+            set_sensor_baudrate(self.ping.port, baud)
```

Comment:
ping device got it taken care of:
https://github.com/bluerobotics/ping-python/blob/deployment/brping/device.py#L38